plugins/modules/cloud/lxd/lxd_virtual_machine.py

```diff
@@ -0,0 +1,744 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2016, Hiroaki Nakamura <hnakamur@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: lxd_virtual_machine
+short_description: Manage LXD Virtual machines
+description:
+  - Management of LXD virtual machines
+author: "Hiroaki Nakamura (@hnakamur)"
+options:
+    name:
+        description:
+          - Name of a virtual machine.
+        type: str
+        required: true
+    architecture:
+        description:
+          - The architecture for the virtual machine (e.g. "x86_64" or "i686").
+            See U(https://github.com/lxc/lxd/blob/master/doc/rest-api.md#post-1)
+        type: str
+        required: false
+    config:
+        description:
+          - 'The config for the virtual machine (e.g. {"limits.cpu": "2"}).
+            See U(https://github.com/lxc/lxd/blob/master/doc/rest-api.md#post-1)'
+          - If the virtual machine already exists and its "config" values in metadata
+            obtained from GET /1.0/virtual-machines/<name>
+            U(https://github.com/lxc/lxd/blob/master/doc/rest-api.md#10containersname)
+            are different, this module tries to apply the configurations.
+          - The keys starting with C(volatile.) are ignored for this comparison when I(ignore_volatile_options=true).
+        type: dict
+        required: false
+    ignore_volatile_options:
+        description:
+          - If set to C(true), options starting with C(volatile.) are ignored. As a result,
+            they are reapplied for each execution.
+          - This default behavior can be changed by setting this option to C(false).
+          - The current default value C(true) is deprecated since community.general 4.0.0,
```

Comment:
You should remove this deprecation and switch to the final state, since there's no reason to stick with the old behavior. (Or even remove this option at all.)

Reply:
Hi @adamcstephens,

Since @hnakamur did not process volatile options in its implementation and we were not sure whether existing playbooks would be affected, we decided to add a deprecated message and change the default behavior in the next releases. Since this is a new module, it is not necessary.